plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2448,7 +2468,55 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
         DeltaLakeInsertTableHandle insertHandle = createInsertHandle(retryMode, handle, inputColumns);
 
         Map<String, DeletionVectorEntry> deletionVectors = loadDeletionVectors(session, handle);
-        return new DeltaLakeMergeTableHandle(handle, insertHandle, deletionVectors);
+        return new DeltaLakeMergeTableHandle(handle, insertHandle, deletionVectors, shallowCloneSourceTableLocation(session, handle));
+    }
+
+    private Optional<String> shallowCloneSourceTableLocation(ConnectorSession session, DeltaLakeTableHandle handle)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        String sourceTableName;
+        try {
+            // The clone commit is the first commit of the cloned table, so set the endVersion to 0
+            TransactionLogTail transactionLogTail = loadNewTail(fileSystem, handle.getLocation(), Optional.empty(), Optional.of(0L), DataSize.ofBytes(0));
+            List<Transaction> transactions = transactionLogTail.getTransactions();
+            if (transactions.isEmpty()) {
+                return Optional.empty();
+            }
+
+            Optional<CommitInfoEntry> cloneCommit = transactions.getFirst().transactionEntries().getEntries(fileSystem)
+                    .map(DeltaLakeTransactionLogEntry::getCommitInfo)
+                    .filter(Objects::nonNull)
+                    .filter(commitInfoEntry -> commitInfoEntry.operation().equals("CLONE"))
+                    .findFirst();
+            if (cloneCommit.isEmpty()) {
+                return Optional.empty();
+            }
+
+            // It's the cloned table
+            Map<String, String> operationParameters = cloneCommit.get().operationParameters();
+            if (!operationParameters.containsKey("source")) {
+                return Optional.empty();
+            }
+
+            sourceTableName = operationParameters.get("source");
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        checkArgument(sourceTableName != null && sourceTableName.contains(".") && sourceTableName.split("\\.").length == 3, "Unexpected source table in operation_parameters: %s", sourceTableName);
```

Comment:
Should we fail if source_table is not composed of 3 elements? In all other cases we just return Optional.empty() from this method.

Reply:
Here is prepare for merge, I am bit concern if we allow this pass will case correctness issue, do you have examples that what kind of data `sourceTableName` could be

Reply:
ok, so now let's only allow the pattern that has parts like `spark.schema.table`, if any other pattern return empty.